Sources/ProjectDescription/SourceFilesList.swift

```diff
@@ -2,6 +2,16 @@ import Foundation
 
 /// A glob pattern configuration representing source files and its compiler flags, if any.
 public struct SourceFileGlob: Codable, Equatable, Sendable {
+    /// Type of the source file.
+    public enum FileType: String, Codable, Sendable {
+        /// File is already present on disk before generating the project.
+        case alwaysPresent
+
+        /// File was generated during the generation of the project, e.g. by a pre-build phase
+        /// script.
+        case generated
```

Comment:
I'm a bit worried about target hashing, as we has no info here to determine if the generated file will change or not.
Should we at least allow here a list of files which hare "inputs" to the generation, so we can use them in hashing?
Something like:
```
case generated(inputs: …)
```

Reply:
Yeah, that's a good point.

When the file is generated using a build phase, the input and output paths should already be in the final hash.

Reply:
Makes sense, but I would add a disclaimer that those should be set, as IIRC they are optional in the build phase

Reply:
Doesn't Xcode complain if you don't set them?